bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/ResourceChangeListenerList.java

```diff
@@ -79,29 +73,23 @@ public String toString() {
 	 * @param mask event types
 	 */
 	public synchronized void add(IResourceChangeListener listener, int mask) {
-		Assert.isNotNull(listener);
```

Comment:
Why is this removed now?

Reply:
listener is final now

Reply:
Sorry, this can't be true - it is argument to the public method.

Reply:
oh my bad